climada/util/coordinates.py

```diff
@@ -1944,6 +1944,41 @@ def align_raster_data(source, src_crs, src_transform, dst_crs=None, dst_resoluti
         raise ValueError(f"Invalid value for conserve: {conserve}")
     return destination, dst_transform
 
+def mask_raster_with_geometry(raster, transform, shapes, nodata=None, **kwargs):
+    """
+    Wrapper for rasterio.mask.mask to allow for in-memory processing.
```

Comment:
Maybe add one sentence to the describtion of the function that spells out what it actually does: "changes values in `raster` that are outside of the provided `shapes` to `nodata`"

Reply:
see https://github.com/CLIMADA-project/climada_python/pull/239/commits/61e39307a0bba3a41e6f81d04a340aa3b842411d